packages-next/fields/src/types/image/Implementation.ts

```diff
@@ -0,0 +1,247 @@
+import { PrismaFieldAdapter, PrismaListAdapter } from '@keystone-next/adapter-prisma-legacy';
+import { ImageData, KeystoneContext, BaseKeystoneList } from '@keystone-next/types';
+import { Implementation } from '../../Implementation';
+import { handleImageData } from './handle-image-input';
+
+export class ImageImplementation<P extends string> extends Implementation<P> {
+  get _supportsUnique() {
+    return false;
+  }
+
+  gqlOutputFields() {
+    return [`${this.path}: ImageFieldOutput`];
+  }
+
+  getGqlAuxTypes() {
+    return [
+      `enum ImageMode {
+        local
+      }
+      input ImageFieldInput {
+        upload: Upload
+        ref: String
+      }
+      enum ImageExtension {
+        jpeg
+        png
+        webp
+        gif
+      }
+      type ImageFieldOutput {
+        mode: ImageMode!
+        id: ID!
+        filesize: Int!
+        width: Int!
+        height: Int!
+        extension: ImageExtension!
+        ref: String!
+        src: String!
+      }`,
+    ];
+  }
+
+  gqlAuxFieldResolvers() {
+    return {
+      ImageFieldOutput: {
+        src(data: ImageData, _args: any, context: KeystoneContext) {
+          if (!context.images) {
+            throw new Error('Image context is undefined');
+          }
+          return context.images.getSrc(data.mode, data.id, data.extension);
+        },
+        ref(data: ImageData, _args: any, context: KeystoneContext) {
+          if (!context.images) {
+            throw new Error('Image context is undefined');
+          }
+          return context.images.getRef(data.mode, data.id, data.extension);
+        },
+      },
+    };
+  }
+  // Called on `User.avatar` for example
+  gqlOutputFieldResolvers() {
+    return {
+      [this.path]: (item: Record<P, any>) => {
+        return item[this.path];
+      },
+    };
+  }
+
+  async validateInput({
+    fieldPath,
+    resolvedData,
+    addFieldValidationError,
+  }: {
+    fieldPath: string;
+    resolvedData: Record<P, any>;
+    addFieldValidationError: (msg: string) => void;
+  }) {
+    if (!resolvedData[fieldPath as P]) return;
+    const img = resolvedData[fieldPath as P];
+    Object.keys(img).forEach(key => {
+      if (!img[key]) {
+        addFieldValidationError(`${fieldPath}.${key} is a non-nullable property`);
+      }
+    });
```

Comment:
This seems wrong, one of the keys has to be nullable in the input?

Reply:
Clarified with @mitchellhamilton, removing this entire check, since we don't want to be running this on the input value